src/offsets.zig

```diff
@@ -80,7 +80,7 @@ pub fn lineSectionLength(tree: ast.Tree, start_index: usize, end_index: usize, e
 
         const codepoint = try std.unicode.utf8Decode(source[i .. i + n]);
 
-        result += 1 + @as(usize, @boolToInt(codepoint >= 0x10000));
+        result += 1 + @boolToInt(codepoint >= 0x10000);
```

Comment:
The coercion here is intentional, @boolToInt will return `u1` and thus the whole expression is of type `u1` and will overflow when the condition is true.

Reply:
Maybe
```zig
result = result + 1 + @boolToInt(codepoint >= 0x10000);
```
is saner here for this reason

Reply:
Thanks for the catch, fixed it.